sdk/datafactory/Azure.ResourceManager.DataFactory/tests/Scenario/DataFactoryDatasetResourceTests.cs

```diff
@@ -27,13 +29,26 @@ public async Task TestSetUp()
             // Create a resource group
             string dataFactoryName = Recording.GenerateAssetName("DataFactory-");
             SubscriptionResource subscription = await Client.GetDefaultSubscriptionAsync();
-            var resourceGroup = await CreateResourceGroup(subscription, "DataFactory-RG-", AzureLocation.WestUS2);
+            _resourceGroup = await CreateResourceGroup(subscription, "DataFactory-RG-", AzureLocation.WestUS2);
             // Create a DataFactory and a LinkedService
-            _dataFactory = await CreateDataFactory(resourceGroup, dataFactoryName);
+            _dataFactory = await CreateDataFactory(_resourceGroup, dataFactoryName);
+            // Create a LinkedService
+            _accessKey = await GetStorageAccountAccessKey(_resourceGroup);
             _linkedServiceName = Recording.GenerateAssetName("LinkedService");
             await CreateLinkedService(_dataFactory, _linkedServiceName, _accessKey);
         }
 
+        [TearDown]
+        public async Task TestTearDown()
+        {
+            // Delete Storage Account ASAP.
+            var list = await _resourceGroup.GetStorageAccounts().GetAllAsync().ToEnumerableAsync();
```

Comment:
As I know, the base class will remove all resource under the test resource group, right?

Reply:
The TestFramework will remove ResourceGroup inside [OneTimeTearDown] after all tests are completed, and this [Teardown] method will remove StorageAccount at the end of each [Test]. I think it is reasonable to remove the invalid resources during the test